### PR TITLE
State law

### PR DESCRIPTION
**Pull request by jrleeman**
Better encapsulate the state evolution relations and make things a bit more independent. Addresses #7 and decoupling.

### CODE REVIEW

File: rsf.py [hunk 1]
```diff
@@ -26,60 +26,107 @@
...
 
-def dieterichState(model):
-    return 1. - model.v * model.theta / model.dc
+class StateRelation(object):
+    def __init__(self, relation):
+        self.b = None
+        self.Dc = None
+        self.state = None
 
+    def velocity_componet(self, system):
```
Comment (dopplershift) [inline rsf.py]: typo: should be `velocity_component`?

File: rsf.py [hunk 2]
```diff
@@ -92,26 +139,32 @@ def solve(self, **kwargs):
...
-        self.results.state1 = wsol[:, 1]
-        self.results.slider_velocity = self.vref * np.exp((self.results.friction - self.mu0 - self.b * np.log(self.vref * self.results.state1 / self.dc)) / self.a)
-        self.results.time = self.model_time
+        self.results.states = wsol[:, 1:]
+        self.results.time = system.model_time
+
+        # Calculate slider velocity after we have solved everything
+        velocity_contribution = 0
+        for i, state_variable in enumerate(system.state_relations):
+            velocity_contribution += state_variable.b * np.log(system.vref * self.results.states[:, i] / state_variable.Dc)
```
Comment (dopplershift) [inline rsf.py]: Any reason you couldn't use the `velocity_component` method here?

File: rsf.py (hunk 1 above)
Comment (jrleeman) [inline rsf.py]: Yes... Oops!

**Comment by jrleeman** [comment]
Ok, I used the velocity_component method again, but had to do a couple of small modifications. I think overall this is a nicer solution though? It does leave an array of the state evolution in each instance of `StateRelation` at the end, which is rather nice.

**Comment by dopplershift** [comment]
Yeah, looks good to me. Some good things I notice:
- There aren't any more numbered variables any more
- RateState does not really do any math any more

**Comment by jrleeman** [comment]
Maybe before we merge we should consider naming `RateSate` to something like `Model`. Maybe it would even make sense to have `self.system = ExternalSystem()` when we create that object? Just trying to think how to make it easiest to interact with, but maybe that's messy.

On numbered variables. Should we have any since the state relations are now getting set as a list?

**Comment by dopplershift** [comment]
Looks like another good step forward. And no, there shouldn't be any numbered vars now. To be clear, I thought those were positives (sorry that was unclear).

**Comment by jrleeman** [comment]
Maybe we're ready to merge this one and hit a few other issues then?

**Pull request merged**